src/imgui/ImGuiWaveViewer.cc

```diff
@@ -123,14 +123,25 @@ static void plotHistogram(std::span<const float> values, float scale_min, float
 }
 
 
+static constexpr std::string_view SHOW_CHIP_PREFIX = "showChip.";
+
 void ImGuiWaveViewer::save(ImGuiTextBuffer& buf)
 {
 	savePersistent(buf, *this, persistentElements);
+	for (const auto& chip: openChips) {
+		buf.appendf("%s%s=%d\n", std::string(SHOW_CHIP_PREFIX).c_str(), chip.c_str(), true);
+	}
 }
 
 void ImGuiWaveViewer::loadLine(std::string_view name, zstring_view value)
 {
-	loadOnePersistent(name, value, *this, persistentElements);
+	if (loadOnePersistent(name, value, *this, persistentElements)) {
+		// already handled
+	} else if (name.starts_with(SHOW_CHIP_PREFIX)) {
+		if (StringOp::stringToBool(value)) {
+			openChips.insert(std::string(name.substr(SHOW_CHIP_PREFIX.size())));
+		}
```

Comment:
See above. with the above suggestion (alternative format) this could become something like:
```
if (name == "showChip") {
    openChips.insert(value);
}
```

There's one thing missing from this code: it can only insert items in the set. So when loading a new "layout" file (so not at startup, but during runtime), the `openChip` set will contain the union of the current set and the loaded values. The fix that you also need to implement the `loadStart()` method (and then clear the set in that method).

Reply:
And this as well.